src/lib/MetamaskDelegatedAccess.ts

```diff
@@ -0,0 +1,82 @@
+import { Wallet } from "ethers"
+import { Buffer } from 'buffer'
+import { encrypt } from "eth-sig-util"
+import { bufferToHex } from 'ethereumjs-util'
+import {StreamrClient} from "streamr-client"
+
+declare global {
+    interface Window { 
+        ethereum: any
+    }
+}
+
+export class MetamaskDelegatedAccess {
+    metamaskAddress?: string 
```

Comment:
Not essential but `import { isAddress } from 'web3-utils'` could help to establish type protection

Reply:
Both `metamaskAddress` and `clientAddress` are assigned from the `window.ethereum` and `ethers.Walllet` objects, respectively. Not sure that adding a check for `isAddress` on those adds much.